modules/govuk_jenkins/templates/jobs/deploy_app_downstream.yaml.erb

```diff
@@ -30,14 +30,30 @@
           fi
 
           # Workaround for our inconsistent repo vs. deployment naming
+          # and multiple / different healthcheck URLs
           case "$TARGET_APPLICATION" in
           <% @applications.keys.each do |app| %>
+            <% default_healthcheck_urls = [
+              "https://#{app}.#{@app_domain}/healthcheck/ready"
+            ] %>
+
             <%= app %>)
               REPO="<%= @applications[app].fetch("repository", app) %>"
+
+              HEALTHCHECK_URLS="<%= @applications[app].fetch(
+                "healthcheck_urls", default_healthcheck_urls
+              ).join(" ")%>"
               ;;
           <% end %>
           esac
 
+          <% if @smokey_pre_check %>
+          # Check healthcheck endpoints return 200 OK
+          for url in $HEALTHCHECK_URLS; do
+            curl $url;
```

Comment:
I think you'll want `--fail` or `--fail-with-body` here, to make `curl` return a nonzero exit code

Reply:
Good spot - I've added `--fail` and tested it works as expected.